src/components/checkbox/checkbox.tsx

```diff
@@ -0,0 +1,39 @@
+import React = require('react');
+
+export interface CheckBoxProps {
```

Comment:
IMHO `interface` is preferable here. `type` usually refer to other type (type alias).